### PR TITLE
Fixes bug that kept deletion code from running 

### PR DESCRIPTION
**Pull request by ehankland**
Fixes bug that kept deletion code from running when there was an exception thrown in the prepare stage

See https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/213

### CODE REVIEW

**Comment by ehankland** [comment]
test this please.

File: perfkitbenchmarker/pkb.py [hunk 1]
```diff
@@ -257,8 +255,8 @@ def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks):
     try:
       with end_to_end_timer.Measure('End to End'):
         if FLAGS.run_stage in [STAGE_ALL, STAGE_PREPARE]:
-          spec = DoPreparePhase(
-              benchmark, benchmark_info, benchmark_name, detailed_timer)
+          spec = benchmark_spec.BenchmarkSpec(benchmark_info)
```
Comment (voellm) [inline perfkitbenchmarker/pkb.py]: Please add a comment on why this separation is important.

By getting the spec you have also caused it to be saved for future cleanup.

**Comment by voellm** [comment]
LGTM

**Comment by ehankland** [comment]
test this please.

**Comment by ehankland** [comment]
test this please.

**Comment by voellm** [comment]
LGTM

**Pull request merged**